### PR TITLE
refactor of initial plugin

### PR DESCRIPTION
**Pull request by jagedn**
- upgrade to nextflow 24
- use carpet lib
- refactor splitParquet
- some validation use cases

This is an initial upgrade of the plugin using an open source library (carpet) to read/write parquet files avoiding the complexity of using low level library.

Carpet can read parquet files as Map but also you can provide a java Record class as projection so the library read only field presents in the Record

This PR contains a `validation` folder, with a few small parquet files, to validate the plugin (tests will be added in next PRs) so you can see in action following these steps:

```
./gradlew copyPluginZip
cd validation
NXF_PLUGINS_DIR=$(pwd)/../build/plugins nextflow run read_raw.nf 
NXF_PLUGINS_DIR=$(pwd)/../build/plugins nextflow run read.nf 
NXF_PLUGINS_DIR=$(pwd)/../build/plugins nextflow run read_complex.nf 
```

### CODE REVIEW

File: plugins/nf-parquet/build.gradle [hunk 1]
```diff
@@ -50,34 +50,26 @@ sourceSets {
...
     testImplementation "io.nextflow:nextflow:$nextflowVersion"
-    testImplementation ("org.codehaus.groovy:groovy-test:3.0.17") { exclude group: 'org.codehaus.groovy' }
-    testImplementation ("cglib:cglib-nodep:3.3.0")
-    testImplementation ("org.objenesis:objenesis:3.1")
-    testImplementation ("org.spockframework:spock-core:2.2-groovy-3.0") { exclude group: 'org.codehaus.groovy'; exclude group: 'net.bytebuddy' }
-    testImplementation ('org.spockframework:spock-junit4:2.2-groovy-3.0') { exclude group: 'org.codehaus.groovy'; exclude group: 'net.bytebuddy' }
-    testImplementation ('com.google.jimfs:jimfs:1.1')
-
+    testImplementation ("org.spockframework:spock-core:2.2-groovy-4.0") { exclude group: 'org.codehaus.groovy'; exclude group: 'net.bytebuddy' }
+    testImplementation ('org.spockframework:spock-junit4:2.2-groovy-4.0') { exclude group: 'org.codehaus.groovy'; exclude group: 'net.bytebuddy' }
```
Comment (pditommaso) [inline plugins/nf-parquet/build.gradle:68]: Let's use spock 2.3 as in nextflow core 

https://github.com/nextflow-io/nextflow/blob/5a37e6177f7a0e02b2af922768a0df5984b07b7b/modules/nextflow/build.gradle#L59-L61

**Comment by jagedn** [comment]
> Great Jorge, the README is still valid?

Yes, the method names are the same (although write method is empty by the moment)

File: plugins/nf-parquet/build.gradle (hunk 1 above)
Comment (pditommaso) [inline plugins/nf-parquet/build.gradle:68]:
```suggestion
    testImplementation ("org.spockframework:spock-core:2.3-groovy-4.0") { exclude group: 'org.codehaus.groovy'; exclude group: 'net.bytebuddy' }
    testImplementation ('org.spockframework:spock-junit4:2.3-groovy-4.0') { exclude group: 'org.codehaus.groovy'; exclude group: 'net.bytebuddy' }
```

**Comment by pditommaso** [comment]
Awesome!

**Comment by jerolba** [comment]
@jagedn FYI: just yesterday I released version 0.2.0:
https://mvnrepository.com/artifact/com.jerolba/carpet-record/0.2.0
Relevant changes :
* upgrade to Parquet 1.14.2 
* review transistive dependencies exclusions if you are not using hadoop (reduces 2.5MB)

**Pull request merged**